docs/DeveloperGuide.md

```diff
@@ -1,11 +1,461 @@
+    1. UI output to user
+
+#### Allocate Sprint Tasks to Members   
+
+Link: [AllocateSprintTaskCommand.java](/src/main/java/seedu/duke/command/sprint/AllocateSprintTaskCommand.java) 
+ 
+Users can allocate Sprint Tasks to Members, indicating that the Tasks are assigned to the selected member to work on during the iteration. 
+
+Prerequisites:    
+1. At least one Task is added to the selected Sprint
+1. At least one Member is added to the Project
+
+Implementation:
+1. UI receive user input
+1. Parser parse user input
+1. Execute AllocateSprintTaskCommand
+    1. On Task
+        1. Add Member ID into membersAllocatedTo
+    1. On Member
+        1. Add Task ID into allocatedTaskIds
+
+
+### Storage
+To make the data persistent and portable, JSON has been chosen as the format for data to be saved to a persistent storage such as storage drives, thumb drives and any other storage medium which stores the program. JSON is also **human-readable** which allows users to directly modify the data file easily. This can be useful in certain scenarios such as fixing the data file in the event of data corruption.
+
+#### Location
+![Figure X: Running the Jar](image/developerguide/savingdata1.png "Running the Jar")  
+_Figure X: Running the Jar_  
+![Figure X: Running in IDE](image/developerguide/savingdata2.png "Running in IDE")  
+_Figure X: Running in IDE_  
+
+As shown in the above diagram, the program will save the data as _"data.json"_. The data file is saved in the _“data/”_ folder that is located in the folder of the program. If you are testing the program using Intellij IDE, the _“data/”_ folder will be in the root of the project folder.  
+When you start the program, the program will load the data file from its respective location and deserialise it into its respective objects. Data will be saved when the program exits or whenever the user makes changes to the program.  
+
+#### Loading Data
+![Figure X: Loading Data](image/developerguide/storage_load.png "Loading Data")  
+
+The program will only load the data file in the persistent storage during the initialisation process of the program. With reference to the sequence diagram above, the flow of the logic is as follows:  
+1. When the user starts the program, it will first call `init()` to initialise the program. 
+2. A `StorageManager` object will be instantiated with the reference to a `ProjectManager` object that is used during load and save operations.
+3. `load()` will read the data file from the persistent storage, deserialise it into a `JsonObject` object and attempt to convert the object into its respective types.  
+  
+The program will exit immediately with an **exit code 1** if any of the conditions are met:
+- Error trying to read the file.
+- Conversion error due to missing properties.
+- Mapping error due to invalid property type (e.g. "name" is expecting a `String` but data read is an `Integer`).
+
+##### Converting and Mapping of JSON to Objects
```

Comment:
Perhaps a sequence diagram can be added to better illustrate the sequence of events happening here.